File: app/src/main/java/android/matrix/ooredoo/qa/problems/java/ReverseString.java
package android.matrix.ooredoo.qa.problems.java;

import java.util.Stack;

public class ReverseString {

    public static String reverse(String str) {
        char[] charArr = str.toCharArray();
        int size = charArr.length;
        Stack<Character> stack = new Stack<Character>();

        int i;
        for(i = 0; i < size; ++i) {
            stack.push(charArr[i]);
        }

        for(i = 0; i < size; ++i) {
            charArr[i] = stack.pop();
        }

        return String.valueOf(charArr);
    }
}

File: app/src/main/java/android/matrix/ooredoo/qa/problems/kotlin/ReverseString.kt
package android.matrix.ooredoo.qa.problems.kotlin

import java.util.*

object ReverseString {

    fun reverse(str: String): String {

        val stack = Stack<Char>()
        val sb  = StringBuilder()

        var i = 0
        while (i < str.length) {
            stack.push(str[i])
            ++i
        }

        i = 0
        while (i < str.length) {
            sb.append(stack.pop())
            ++i
        }

        return sb.toString();
    }
}
